src/main/java/uk/gov/hmcts/ccd/domain/service/common/AttributeBasedAccessControlService.java

```diff
@@ -0,0 +1,278 @@
+                                                              Set<String> userRoles,
+                                                              Predicate<AccessControlList> access) {
+        applyAccessProfileRules(caseFieldDefinitions);
+
+        return super.filterCaseFieldsByAccess(caseFieldDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public CaseUpdateViewEvent filterCaseViewFieldsByAccess(CaseUpdateViewEvent caseEventTrigger,
+                                                            List<CaseFieldDefinition> caseFieldDefinitions,
+                                                            Set<String> userRoles,
+                                                            Predicate<AccessControlList> access) {
+        applyAccessProfileRules(caseFieldDefinitions);
+
+        return super.filterCaseViewFieldsByAccess(caseEventTrigger,
+            caseFieldDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public List<AuditEvent> filterCaseAuditEventsByReadAccess(List<AuditEvent> auditEvents,
+                                                              List<CaseEventDefinition> caseEventDefinitions,
+                                                              Set<String> userRoles) {
+        return super.filterCaseAuditEventsByReadAccess(auditEvents,
+            caseEventDefinitions,
+            userRoles);
+    }
+
+    @Override
+    public List<CaseStateDefinition> filterCaseStatesByAccess(List<CaseStateDefinition> caseStateDefinitions,
+                                                              Set<String> userRoles,
+                                                              Predicate<AccessControlList> access) {
+        return super.filterCaseStatesByAccess(caseStateDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public List<CaseEventDefinition> filterCaseEventsByAccess(List<CaseEventDefinition> caseEventDefinitions,
+                                                              Set<String> userRoles,
+                                                              Predicate<AccessControlList> access) {
+        return super.filterCaseEventsByAccess(caseEventDefinitions,
+            userRoles,
+            access);
+    }
+
+    @Override
+    public CaseViewActionableEvent[] filterCaseViewTriggersByCreateAccess(
+        CaseViewActionableEvent[] caseViewTriggers,
+        List<CaseEventDefinition> caseEventDefinitions,
+        Set<String> userRoles) {
+        return super.filterCaseViewTriggersByCreateAccess(caseViewTriggers,
+            caseEventDefinitions,
+            userRoles);
+    }
+
+    private void applyAccessProfileRules(List<CaseFieldDefinition> caseFieldDefinitions) {
+        Optional<CaseTypeDefinition> caseTypeDefinition = getCaseTypeDefinition(caseFieldDefinitions);
```

Comment:
Can we move this into the `if`? No need to get the definition if we then won't do anything with it.

Reply:
that should be 
if (caseTypeDefinition.isPresent()) {
            applyAccessProfileRules(caseTypeDefinition.get());
        }